app/src/test/java/com/glima/ilovecats/detail/BreedDetailViewModelTest.kt

```diff
@@ -0,0 +1,66 @@
+package com.glima.ilovecats.detail
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.glima.domain.business.model.BreedImage
+import com.glima.domain.business.usecase.LoadRandomImageForBreedingUseCase
+import com.glima.ilovecats.BreedVO
+import com.glima.ilovecats.getOrAwaitValue
+import io.mockk.MockKAnnotations
+import io.mockk.coEvery
+import io.mockk.impl.annotations.MockK
+import junit.framework.Assert.assertEquals
+import kotlinx.coroutines.flow.flow
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class BreedDetailViewModelTest {
+
+    @get:Rule
+    var rule = InstantTaskExecutorRule()
+
+    @MockK
+    lateinit var useCase: LoadRandomImageForBreedingUseCase
+
+    lateinit var viewModel: BreedDetailViewModel
+
+    private val breeVO = BreedVO(
+        id = "abys",
+        name = "Abyssinian",
+        intelligence = 5,
+        affectionLevel = 5,
+        isRare = true,
+        lifeSpan = "immortal",
+        description = "cute cat breed <3"
+    )
+
+    private val breedImage = BreedImage(
+        id = "1234",
+        url = "some_url"
+    )
+
+    @Before
+    fun setupTestEnvironment() {
+        MockKAnnotations.init(this)
+
+        viewModel = BreedDetailViewModel(
+            loadRandomImageUseCase = useCase,
+            breedVO = breeVO
+        )
+    }
+
+    @Test
+    fun loadImage_loadsNewRandomBreedImage() {
```

Comment:
Unfortunately, my test didn't work 😢   for some reason ```value``` isn't being assigned :/

Reply:
I figured out why my test isn't working. My mock wasn't correct, The observable wasn't being emitted before, so I replaced the Mockk return to ```coEvery { useCase.execute(breeVO.id) } returns liveData { emit(breedImage) }.asFlow()
``` and run the test inside a ```runblockingTest``` and it worked. :)